File: JavaScript/7js 2/indexOf.js
/*
|--------------------------------------------------------------------------
| Fun with lists: indexOf
|--------------------------------------------------------------------------
|
*/

function indexOf(head, value) {
   for (let i = 0; head; head = head.next, i++) {
      if (head.data === value) return i;
   }
   return -1;
}


function indexOf(head, value, i = 0) {
   return head == null ? -1 : head.data === value ? i : indexOf(head.next, value, i + 1);
}


function indexOf(head, value) {
   let i = 0;
   while (head) {
      if (head.data === value) return i;
      head = head.next;
      i++;
   }
   return -1;
}

File: Python/7py 2/index_of.py
"""
|--------------------------------------------------------------------------
| Fun with lists: indexOf
|--------------------------------------------------------------------------
|
"""

def index_of(head, value):
   i = 0
   while head:
      if head.data == value: return i
      head = head.next
      i += 1
   return -1


def index_of(node, value, i=0):
   return -1 if node is None else i if node.data == value else index_of(node.next, value, i+1)


def index_of(head, value): 
   i=0
   while head and head.data!=value:
      i+=1
      head=head.next
   return i if head else -1